test/test_nn.py

```diff
@@ -3198,28 +3198,33 @@ def verify_reduction_scalars(input, reduction, output):
                         verify_reduction_scalars(input, reduction, output)
 
     def test_multihead_attention(self):
-        def _scaled_dot_attn_ref(Q, K, V, dims, unseen_mask=False, src_lengths=None):
+        def _scaled_dot_attn_ref(Q, K, V, dims, unseen_mask=None, src_lengths=None, 
+                                 attn_mask=None, add_zero_attn=False):
             """ Numpy-based reference implementation of scaled dot attention
             for testing"""
+
             QKT = _batchmatmul(
                 Q,
                 np.transpose(K, axes=[0, 1, 3, 2])
                 / np.sqrt(dims[3], dtype=np.float32),  # divide by sqrt(d_head)
             )
-            if unseen_mask or src_lengths is not None:
-                b1, b2, s1, s2 = QKT.shape
+            b1, b2, s1, s2 = QKT.shape
+            if unseen_mask is not None or src_lengths is not None:
                 # assert s1 == s2
                 for i in range(b1):
                     for j in range(b2):
                         for m in range(s1):
                             for n in range(s2):
-                                if unseen_mask and n > m:
+                                if unseen_mask[m][n] == 0:
```

Comment:
Why does this change?

Reply:
I re-define unseen_mask and an actual mask tensor is passed to unseen_mask. It's not boolean.